modules/platform/nuxeo-easyshare-core/src/main/java/org/nuxeo/easyshare/EasyShare.java

```diff
@@ -185,18 +292,118 @@ private boolean checkIfShareIsValid(DocumentModel docShare) {
+
+        // Consider a default authorization if no password is not required in easyshare folder metadata (default easyshare feature) 
+        if (BooleanUtils.isNotTrue(docAccessCodeRequired)){
+            // AccessCode not required
+            return true;
+        } else if (StringUtils.isNotBlank(accessCode) && accessCode.equals(docAccessCode)) {
+            // AccessCode provided and mathing the accessCode defined in Easyshare folder
+            return true;
+        } else {
+            log.error("accessCode is empty or does not match for share: " + docShare.getTitle() + " (" + docShare.getId() + ")");
+            // Email notification
+            Map<String, Object> mail = new HashMap<>();
+            sendNotification("easyShareWrongAccessCode", docShare, mail);
+            return false;
+        }
+    }
+
     @Path("{shareId}/{folderId}")
     @GET
-    public Object getFolderListing(@PathParam("shareId") String shareId, @PathParam("folderId") final String folderId,
-            @DefaultValue(DEFAULT_PAGE_INDEX) @QueryParam("p") final Long pageIndex) {
-        return buildUnrestrictedRunner(folderId, pageIndex).runUnrestricted(shareId);
+    public Object getFolderListing(@PathParam("shareId") String shareId,
+            @PathParam("folderId") final String folderId,
+            @DefaultValue(DEFAULT_PAGE_INDEX) @QueryParam("p") final Long pageIndex, 
+            @QueryParam("a") final String encryptedAccessCode,
+            @QueryParam("s") final String encryptedSalt,
+            @QueryParam("j") final String encryptedJSessionId) {
+
+    	String decryptedAccessCode = null;
+        String decryptedJSessionId = null;
+        
+        if (StringUtils.isNotBlank(encryptedAccessCode) && StringUtils.isNotBlank(encryptedSalt)) {
+            String decryptedSalt = EasyShareEncryptionUtil.decodeWithSaltString(encryptedSalt,
+                                                         EasyShareEncryptionUtil.getDefaultSaltAsString());
+            decryptedAccessCode = EasyShareEncryptionUtil
+                                                        .decodeWithSaltString(encryptedAccessCode, decryptedSalt);
+            decryptedJSessionId = EasyShareEncryptionUtil
+                            .decodeWithSaltString(encryptedJSessionId, decryptedSalt);
+        }
+        return buildUnrestrictedRunner(decryptedJSessionId, decryptedAccessCode, folderId, pageIndex).runUnrestricted(shareId);
+    }
+
+    @Path("{shareId}/{folderId}")
+    @POST
+    public Object postFolderListing(@PathParam("shareId") String shareId,
+            @PathParam("folderId") final String folderId,
+            @DefaultValue(DEFAULT_PAGE_INDEX) @FormParam("p") Long pageIndex,
+            @FormParam("accessCode") final String accessCode) {
+        return buildUnrestrictedRunner(getRequestJSESSIONID(), accessCode, folderId, pageIndex).runUnrestricted(shareId);
```

Comment:
Why not getting the JSessionId from the parameters like it is done on the GET endpoint?

Or, why getting the JSessionId from the parameters in the GET endpoint and not from the cookie?

Reply:
In fact, I don't understand why we're having POST endpoint for get folder listing, get blob stream, etc...

Reply:
Can check my answer : [](https://github.com/nuxeo/nuxeo/pull/4520#discussion_r569420929)